test/test_jit.py

```diff
@@ -3360,8 +3370,17 @@ def func():
             print(b)
         ''')
 
-        # test that equal tuples correctly work with node hashing
-        for tup in map(lambda x: "(" + x + ",)", constants):
+        # generate dicts with built-in types (excluding torch.Tensor)
+        xprod = itertools.product(constants, constants)
+
```

Comment:
Can you add the dictionary equivalent here: 
```
        for type in ["Tensor", "str", "int", "float", "bool"]:
            constants.append("torch.jit.annotate(List[ " + type + "], [])")
```

I suspect this might have caught the bug you have right now.